### PR TITLE
Remove unnecessary double download of same dataset during test

### PR DESCRIPTION
**Pull request by MaximeSongIdris**
### Checks to be done before submitting your PR
- [x] `python3 -m pytest tests/` runs successfully.
- [x] `black .` runs successfully.
- [x] `make html` runs successfully (in the `docs/` directory).
- [x] Updated docstrings related to the changes (as applicable).
- [ ] Added an entry to the [CHANGELOG.rst](../CHANGELOG.rst).

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -36,12 +36,12 @@ venv
 build
 deepinv.egg-info
 dist
-datasets
+examples/datasets
```
Comment (tachella) [inline .gitignore:39]: Is this removing the examples from git tracking?

File: deepinv/datasets/div2k.py [hunk 2]
```diff
@@ -30,81 +50,70 @@ class DIV2K(torch.utils.data.Dataset):
...
 
+        # download a split of the dataset, we check first that this split isn't already downloaded
+        if not self._verify_split_dataset_integrity():
+            if download:
+                if not os.path.isdir(self.root):
+                    os.makedirs(self.root)
+                if os.path.exists(self.img_dir):
+                    raise ValueError(
+                        f"""
+                    The {self.mode} folder already exists,
```
Comment (tachella) [inline deepinv/datasets/div2k.py:80]: can you print all the string in one line, without any space at the beginning?

File: .gitignore [hunk 3]
```diff
@@ -36,12 +36,12 @@ venv
...
 wandb/
 kernels
 measurements/
 degradations/
-datasets/
+examples/*/datasets
```
Comment (tachella) [inline .gitignore:44]: same question here?

**Review by tachella — CHANGES_REQUESTED** [review]
the PR looks good, but I think there might be an issue with the gitignore.

File: .gitignore (hunk 1 above)
Comment (MaximeSongIdris) [reply .gitignore:39]: initially it removed every "datasets" folder in the git tracking
notably the "deepinv.datasets" folder
and I changed it to examples/datasets, which contains raw dataset

Comment (MaximeSongIdris) [reply .gitignore:39]: i though that the initial behavior was intended for "examples/datasets" -> do i need to change this line ?

File: .gitignore (hunk 3 above)
Comment (MaximeSongIdris) [reply .gitignore:44]: same as above, 
I inferred that the expected behavior was : 
ignoring all "datasets" subfolders in "examples" folder

File: deepinv/datasets/div2k.py (hunk 2 above)
Comment (MaximeSongIdris) [reply deepinv/datasets/div2k.py]: Done :
every multi line strings  ->  one-line string

**Review by tachella — APPROVED** [review]
thanks for the PR, looks good to me!

**Pull request merged**